report_custom_filename/controllers/reports.py

```diff
@@ -17,10 +17,10 @@ def index(self, action, token):
         context = dict(http.request.context)
         context.update(action["context"])
         report_xml = http.request.session.model('ir.actions.report.xml')
```

Comment:
Better to use `htt.request.env['ir.actions.report.xml']` for using new API and don't change the rest.